week2/homework/src/index.js

```diff
@@ -1,3 +1,68 @@
 'use strict';
 
-// TODO: Write the homework code in this file
+let fs = require('fs');
+let args = process.argv.slice(2);
+// console.log(args);
+
+let command = args[0];
+let todoItem = args[1];
+
+if (command === undefined || 'help') {
+  help(todoItem);
+} else if (command === 'add') {
+  add(todoItem);
+} else if (command === 'list') {
+  list(todoItem);
+} else if (command === 'remove') {
+  remove(todoItem);
+} else if (command === 'reset') {
+  reset(todoItem);
+} else {
+  console.log(`${command} is not recognized as a command.
+  Check the spelling of the name, or if a path was included,
+  verify that the path is correct and try again.
+  Or use help command`);
+}
+
+function list() {
+  fs.readFile('./todoList.txt', 'utf8', (error, todoList) => {
+    if (error) {
+      if (error.code === 'ENOENT') {
+        console.log('no data found');
+      } else console.error(error);
+    } else {
+      console.log(todoList);
+    }
+  });
+}
+
+function add(todoItem) {
+  fs.appendFile('./todoList.txt', todoItem + '\n', error => {
+    console.error(error);
```

Comment:
The `error` should be checked, and logged only if it is present. So the reporting to user will be proper.
Like

    if(error) return console.error(error);
    return console.log('Todo item is added');

Reply:
that is ok, thanks